servo/connectors/olas/fastpath.py

```diff
@@ -0,0 +1,146 @@
+# This Module is a simplified online prediction model that predicts immediate
+# future traffic based on traffic history of short term and high frequency.
+
+import numpy as np
+import pandas as pd
+import pygam
+
+
+class Fastpath(object):
+    '''
+    Online prediction model that predicts immediate future traffic.
+
+    param:
+        - data: history data used to instantiate Fastpath model
+        - foresight: in mins, denotes how much in advance the prediction will be,
+                     and also what frequency used to downsample raw data.
+                     Should be larger than the cadence of received measurements.
+        - window_size: window size used to smooth resampled time series
+    '''
+
+    def __init__(self, data, foresight=1, window_size=5):
+        self.data_lake = self.build_data_lake(data)
+        self.window_size = int(window_size)
+        self.foresight = int(foresight)
+        if self.foresight < 1:
+            raise ValueError(
+                "Fastpath Argument Error: "
+                "Lowest resolution supported for prediction is 1 min. "
+                "Please modidy argument for foresight.")
+
+        self.resample()
+        # Window_size indicats half width of intended sliding window,
+        # thus 4 in below condition is chosen to guarantee at least
+        # half of resampled data could be applied with hampel_filter.
+        if len(self.traffic) < 4 * self.window_size:
+            raise ValueError(
+                "Fastpath Argument Error: "
+                "Initial data provided are not commensurate with window size. "
+                "Please either decrease window size or increase data quantity. "
+                "Suggested time span for initial data should be at least 4*window_size*foresight mins.")
```

Comment:
This constraint says "Suggest" but its actually enforced as a validation. Needs to be documented and explained.